src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java

```diff
@@ -12,23 +18,48 @@
-		this.driverManager = driverManager;
-	}
+    public SelectTestFigureOptionListener(DriverManager driverManager) {
+        this.driverManager = driverManager;
+    }
 
-	@Override
-	public void actionPerformed(ActionEvent e) {
+    @Override
+    public void actionPerformed(ActionEvent e) {
 
-		if(e.getActionCommand().equals(Figures.FIGURE_JOE_1.getCommand())){
-			FiguresJoe.figureScript1(driverManager.getCurrentDriver());
-		}else if(e.getActionCommand().equals(Figures.FIGURE_JOE_2.getCommand())){
-			FiguresJoe.figureScript2(driverManager.getCurrentDriver());
-		}
-		else if(e.getActionCommand().equals(Figures.FIGURE_JANE.getCommand())){
-			FiguresJane.figureScript(new Job2dDriverAbstractAdapter(driverManager.getCurrentDriver()));
-		}
-	}
+        if (e.getActionCommand().equals(Figures.FIGURE_JOE_1.getCommand())) {
+            FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+        } else if (e.getActionCommand().equals(Figures.FIGURE_JOE_2.getCommand())) {
+            FiguresJoe.figureScript2(driverManager.getCurrentDriver());
+        } else if (e.getActionCommand().equals(Figures.FIGURE_JANE.getCommand())) {
+            FiguresJane.figureScript(new Job2dDriverAbstractAdapter(driverManager.getCurrentDriver()));
+        } else if (e.getActionCommand().equals(Figures.FIGURE_CUSTOM.getCommand())) {
+            ComplexCommand complexCommand = new ComplexCommand();
+
+            complexCommand.addCommand(new SetPositionCommand(30, 30));
+            complexCommand.addCommand(new OperateToCommand(-30, 30));
+            complexCommand.addCommand(new OperateToCommand(-30, -30));
+            complexCommand.addCommand(new OperateToCommand(30, -30));
+            complexCommand.addCommand(new OperateToCommand(30, 30));
+            complexCommand.execute(driverManager.getCurrentDriver());
+        }else if (e.getActionCommand().equals(Figures.FIGURE_CUSTOM_TR.getCommand())) {
+            ComplexCommand complexCommand = new ComplexCommand();
+
+            complexCommand.addCommand(new SetPositionCommand(90,0 ));
+            complexCommand.addCommand(new OperateToCommand(0, 90));
+            complexCommand.addCommand(new OperateToCommand(-90, 0));
+            complexCommand.addCommand(new OperateToCommand(90, 0));
+            complexCommand.execute(driverManager.getCurrentDriver());
+        }else if (e.getActionCommand().equals(Figures.FIGURE_CUSTOM_RT.getCommand())) {
+            ComplexCommand complexCommand = new ComplexCommand();
+
+            complexCommand.addCommand(new SetPositionCommand(60, 30));
+            complexCommand.addCommand(new OperateToCommand(-60, 30));
+            complexCommand.addCommand(new OperateToCommand(-60, -30));
+            complexCommand.addCommand(new OperateToCommand(60, -30));
+            complexCommand.addCommand(new OperateToCommand(60, 30));
+            complexCommand.execute(driverManager.getCurrentDriver());
```

Comment:
This is not a factory. A factory should have methods that return a driver command (not execute it). Missing point 3.1.5.